### PR TITLE
nimble/host: Add HCI Commands/ events for connection subrating

### PR DESCRIPTION
**Pull request by rahult-github**
This PR adds support for HCI commands and events in stack for Connection subrating feature.

### CODE REVIEW

**Comment by rahult-github** [comment]
Hi @sjanc , this PR attempts to add host side support for Connection Subrating feature. Can you please take a look ?

File: nimble/host/include/host/ble_gap.h [hunk 1]
```diff
@@ -2167,6 +2194,44 @@ int ble_gap_set_prefered_default_le_phy(uint8_t tx_phys_mask,
...
+ *                          after a packet containing PDU with non-zero length field
+ *                          is sent or received in request by a peripheral.
+ *
+ * @param supervision_timeout Max supervision timeout allowed in request by a peripheral
+ */
+int ble_gap_set_default_subrate(uint16_t subrate_min, uint16_t subrate_max, uint16_t max_latency,
+                                uint16_t cont_num, uint16_t supervision_timeout);
+
+/**
+ * Set Subrate Request
```
Comment (andrzej-kaczmarek) [inline nimble/host/include/host/ble_gap.h:2215]: 'subrate request' (without 'set')

File: nimble/host/include/host/ble_gap.h [hunk 2]
```diff
@@ -2167,6 +2194,44 @@ int ble_gap_set_prefered_default_le_phy(uint8_t tx_phys_mask,
...
+ *
+ * @param conn_handle       Connection Handle of the ACL.
+ * @param subrate_min       Min subrate factor allowed in request by a peripheral
```
Comment (andrzej-kaczmarek) [inline nimble/host/include/host/ble_gap.h:2218]: subrate request can be initiated by both central and peripheral so descriptions are not accurate (c&p...)

**Review by andrzej-kaczmarek — CHANGES_REQUESTED** [review]

File: nimble/host/include/host/ble_gap.h (hunk 2 above)
Comment (rahult-github) [reply nimble/host/include/host/ble_gap.h]: Thank you. Have modified now.

File: nimble/host/include/host/ble_gap.h (hunk 1 above)
Comment (rahult-github) [reply nimble/host/include/host/ble_gap.h]: Thank you. Have modified now.

**Review by sjanc — COMMENTED** [review]
in general looks good to me but implementation should enabled only if enabled by config option nimble/syscfg.yml  eg BLE_SUBRATE

similar to eg periodic sync transfer support  (MYNEWT_VAL(BLE_PERIODIC_ADV_SYNC_TRANSFER))

**Comment by rahult-github** [comment]
> in general looks good to me but implementation should enabled only if enabled by config option nimble/syscfg.yml eg BLE_SUBRATE
> 
> similar to eg periodic sync transfer support (MYNEWT_VAL(BLE_PERIODIC_ADV_SYNC_TRANSFER))

Thanks for this input. Yes, i missed making it configurable. Have added the change now.

**Comment by sjanc** [comment]
looks like change to nimble/syscfg.yml is missing

**Comment by andrzej-kaczmarek** [comment]
LL will use `BLE_LL_CFG_FEAT_CONN_SUBRATING` so syscfg flags should be called `BLE_CONN_SUBRATING` to be consistent

**Comment by rahult-github** [comment]
The changes for macro name change and including in syscfg.xml are now pushed. Also changed the coding style issue reported in previous commit.

**Review by sjanc — APPROVED** [review]

**Review by andrzej-kaczmarek — APPROVED** [review]

**Pull request merged**